### PR TITLE
Improve eodatasets3 performance

### PR DESCRIPTION
**Pull request by supermarkion**
In this PR, we are doing:
1. Remove the nodata values from Config file. The metadata will extract them from output dataset nodata field
2. Avoid the actually data loading from XArray.Dataset (e.g. dataset[band]. values). Then the eodatasets3 metadata files section can be lightweight.

### CODE REVIEW

File: libs/stats/odc/stats/_cli_run.py [hunk 1]
```diff
@@ -36,7 +36,7 @@
     help="Mark outputs for public access (default: no)",
 )
 @click.option(
-    "--apply_eodatasets3/--not_apply_eodatasets3",
+    "--apply_eodatasets3",
```
Comment (omad) [inline libs/stats/odc/stats/_cli_run.py:39]:
```suggestion
    "--apply-eodatasets3",
```
Click will still use the same variable name, but it's prettier and more consistent with other application options.

File: libs/stats/odc/stats/model.py [hunk 2]
```diff
@@ -400,11 +401,11 @@ def render_assembler_metadata(
                 # when we pass grid, the eodatasets will not load file from path
                 dataset_assembler.note_measurement(band,
                                                     path,
-                                                    pixels=output_dataset[band].values.reshape([self.geobox.shape[0], self.geobox.shape[1]]),
+                                                    pixels=numpy.zeros((self.geobox.shape[0], self.geobox.shape[1])),
```
Comment (omad) [inline libs/stats/odc/stats/model.py:404]: Why does `.note_measurement()` even require a `pixels` argument?

**Review by kieranricardo — APPROVED** [review]
LGTM :)

**Comment by supermarkion** [comment]
Improve the eodatasets3 performance by changing data access pattern.

File: libs/stats/odc/stats/model.py (hunk 2 above)
Comment (supermarkion) [reply libs/stats/odc/stats/model.py:404]: In this method, it needs the pixel values to check nodata: https://github.com/GeoscienceAustralia/eo-datasets/blob/3c38a5a740980389452884ba7a53fdee79f7cf1a/eodatasets3/images.py#L308

The calling orders are: 
1. https://github.com/GeoscienceAustralia/eo-datasets/blob/eodatasets3/eodatasets3/assemble.py#L736
2. https://github.com/GeoscienceAustralia/eo-datasets/blob/3c38a5a740980389452884ba7a53fdee79f7cf1a/eodatasets3/images.py#L278
3. https://github.com/GeoscienceAustralia/eo-datasets/blob/3c38a5a740980389452884ba7a53fdee79f7cf1a/eodatasets3/images.py#L297
4. https://github.com/GeoscienceAustralia/eo-datasets/blob/3c38a5a740980389452884ba7a53fdee79f7cf1a/eodatasets3/images.py#L308

**Pull request merged**